Library/Homebrew/services/cli.rb

```diff
@@ -237,7 +246,12 @@ def self.kill(targets, verbose: false)
             if System.systemctl?
               System::Systemctl.quiet_run("stop", service.service_name)
             elsif System.launchctl?
-              quiet_system System.launchctl, "stop", "#{System.domain_target}/#{service.service_name}"
+              domain_targets = System.candidate_domain_targets
+              while service.pid? && !domain_targets.empty?
```

Comment:
Does `--no-wait` need handled here? Same concern above about this potentially blocking forever.